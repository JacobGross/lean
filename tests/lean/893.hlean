example {A : Type} (f : A â†’ A) (a : A) (H : f a = a) : unit :=
begin
  induction H, -- this should raise an error
  exact unit.star
end
