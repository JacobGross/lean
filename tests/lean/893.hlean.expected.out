893.hlean:3:2: error:invalid 'induction' tactic, invalid 'induction' tactic, argument #3 of major premise 'H' type is an index, but it occurs in previous arguments
proof state:
A : Type,
f : A → A,
a : A,
H : f a = a
⊢ unit
893.hlean:5:0: error: don't know how to synthesize placeholder
A : Type,
f : A → A,
a : A,
H : f a = a
⊢ unit
893.hlean:5:0: error: failed to add declaration 'example' to environment, value has metavariables
remark: set 'formatter.hide_full_terms' to false to see the complete term
  ?M_1
